node/chainSimulator/configs/configs.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return nil, err
+	}
+
+	gasScheduleName, err := GetLatestGasScheduleFilename(configs.ConfigurationPathsHolder.GasScheduleDirectoryName)
+	if err != nil {
+		return nil, err
+	}
+
+	configs.GeneralConfig.SmartContractsStorage.DB.Type = string(storageunit.MemoryDB)
+	configs.GeneralConfig.SmartContractsStorageForSCQuery.DB.Type = string(storageunit.MemoryDB)
+	configs.GeneralConfig.SmartContractsStorageSimulate.DB.Type = string(storageunit.MemoryDB)
+
+	return &ArgsConfigsSimulator{
+		Configs:               configs,
+		ValidatorsPrivateKeys: privateKeys,
+		GasScheduleFilename:   gasScheduleName,
+	}, nil
+}
+
+func generateValidatorsKeyAndUpdateFiles(tb testing.TB, configs *config.Configs, numOfShards uint32, address string) ([]crypto.PrivateKey, []crypto.PublicKey) {
+	blockSigningGenerator := signing.NewKeyGenerator(mcl.NewSuiteBLS12())
+
+	nodesSetupFile := configs.ConfigurationPathsHolder.Nodes
+	nodes := &sharding.NodesSetup{}
+	err := core.LoadJsonFile(nodes, nodesSetupFile)
+	require.Nil(tb, err)
+
+	nodes.ConsensusGroupSize = 1
+	nodes.MinNodesPerShard = 1
+	nodes.MetaChainMinNodes = 1
+	nodes.MetaChainConsensusGroupSize = 1
+	nodes.InitialNodes = make([]*sharding.InitialNode, 0)
+
+	privateKeys := make([]crypto.PrivateKey, 0, numOfShards+1)
+	publicKeys := make([]crypto.PublicKey, 0, numOfShards+1)
+	for idx := uint32(0); idx < numOfShards+1; idx++ {
+		sk, pk := blockSigningGenerator.GeneratePair()
+		privateKeys = append(privateKeys, sk)
+		publicKeys = append(publicKeys, pk)
+
+		pkBytes, errB := pk.ToByteArray()
+		require.Nil(tb, errB)
+
+		nodes.InitialNodes = append(nodes.InitialNodes, &sharding.InitialNode{
+			PubKey:  hex.EncodeToString(pkBytes),
+			Address: address,
+		})
+	}
+
+	marshaledNodes, err := json.Marshal(nodes)
+	require.Nil(tb, err)
+
+	err = os.WriteFile(nodesSetupFile, marshaledNodes, 0644)
+	require.Nil(tb, err)
+
+	return privateKeys, publicKeys
+}
+
+func generateValidatorsPem(validatorsFile string, publicKeys []crypto.PublicKey, privateKey []crypto.PrivateKey) error {
+	validatorPubKeyConverter, err := pubkeyConverter.NewHexPubkeyConverter(96)
+	if err != nil {
+		return err
+	}
+
+	buff := bytes.Buffer{}
+	for idx := 0; idx < len(publicKeys); idx++ {
+		publicKeyBytes, errA := publicKeys[idx].ToByteArray()
+		if errA != nil {
+			return errA
+		}
+
+		pkString, errE := validatorPubKeyConverter.Encode(publicKeyBytes)
+		if errE != nil {
+			return errE
+		}
+
+		privateKeyBytes, errP := privateKey[idx].ToByteArray()
+		if errP != nil {
+			return errP
+		}
+
+		blk := pem.Block{
+			Type:  "PRIVATE KEY for " + pkString,
+			Bytes: []byte(hex.EncodeToString(privateKeyBytes)),
+		}
+
+		err = pem.Encode(&buff, &blk)
+		if err != nil {
+			return err
+		}
+	}
+
+	return os.WriteFile(validatorsFile, buff.Bytes(), 0644)
+}
+
+func modifyFile(fileName string, f func(i []byte) ([]byte, error)) error {
```

Comment:
not a big fan of providing a function pointer. Do we really need this or we can simplify the function signature?

Reply:
remove this function